server/service/src/invoice/outbound_return/update/generate.rs

```diff
@@ -0,0 +1,107 @@
+use chrono::Utc;
+use repository::{InvoiceRow, InvoiceRowStatus, StockLineRow, StorageConnection};
+
+use crate::invoice::common::{
+    generate_batches_total_number_of_packs_update, InvoiceLineHasNoStockLine,
+};
+
+use super::{UpdateOutboundReturn, UpdateOutboundReturnError, UpdateOutboundReturnStatus};
+
+pub struct GenerateResult {
+    pub updated_return: InvoiceRow,
+    pub stock_lines_to_update: Option<Vec<StockLineRow>>,
+}
+
+pub fn generate(
+    connection: &StorageConnection,
+    input: UpdateOutboundReturn,
+    existing_return: InvoiceRow,
+) -> Result<GenerateResult, UpdateOutboundReturnError> {
+    let mut updated_return = existing_return.clone();
+
+    updated_return.comment = input.comment.or(existing_return.comment.clone());
```

Comment:
Should we also add updating `colour` here, or do that later?

Reply:
Not sure about these, maybe create a follow up issue? I don't know how the on_hold status is applied for other invoices, I'd assume we'd follow a similar pattern unless returns shouldn't ever go `on_hold`?

Colour sounds interesting, is it being set anywhere else in the code base? or just a legacy mSupply thing?